core/trino-main/src/main/java/io/trino/testing/TestingNodeManager.java

```diff
@@ -83,7 +101,7 @@ public Set<Node> getAllNodes()
     @Override
     public Set<Node> getWorkerNodes()
     {
-        return nodes;
+        return nodes.stream().filter(n -> scheduleOnCoordinator || !n.isCoordinator()).collect(toImmutableSet());
```

Comment:
rename n to node, wrap lines.

if scheduleOnCoordinator === true, there is no node filtering needed as the filter predicate is always true

```
if (scheduleOnCoordinator) {
	return nodes;
}
```
```
nodes.stream()
.filter(node -> !node.isCoordinator())
.collect(toImmutableSet());
```